spec/components/holiday_usage_table_component_spec.rb

```diff
@@ -0,0 +1,221 @@
+      it 'returns true' do
+        expect(component.can_compare_holiday_usage?(holiday_4)).to be true
+      end
+    end
+
+    context 'when we have partial data for current holiday' do
+      let(:data_end_date) { final_holiday_end - 2.days }
+
+      it 'returns false' do
+        expect(component.can_compare_holiday_usage?(holiday_4)).to be false
+      end
+    end
+  end
+
+  describe '.within_school_period?' do
+    context 'when we are past holiday' do
+      it 'returns true' do
+        expect(component.within_school_period?(holiday_1)).to be false
+      end
+    end
+
+    context 'when we have partial data for current holiday' do
+      let(:data_end_date) { final_holiday_end - 2.days }
+
+      it 'returns false' do
+        expect(component.within_school_period?(holiday_4)).to be true
+      end
+    end
+  end
+
+  describe '.average_daily_usage' do
+    it 'returns expected result' do
+      usage = CombinedUsageMetric.new(kwh: 100.0, £: 100.0, co2: 100.0)
+      expect(component.average_daily_usage(usage, holiday_3)).to eq(usage.kwh / holiday_3.days)
+    end
+  end
+
+  describe '.current_holiday_row' do
+    it 'yields expected values' do
+      component.current_holiday_row(holiday_4) do |label, holiday, usage, average_daily_usage|
+        expect(label).to eq('')
+        expect(holiday).to eq(holiday_4)
+        expect(usage).to eq(holiday_usage[holiday_4].usage)
+        expect(average_daily_usage).to eq(holiday_usage[holiday_4].usage.kwh / holiday_4.days)
+      end
+    end
+  end
+
+  describe '.previous_holiday_row' do
+    it 'yields expected values' do
+      component.previous_holiday_row(holiday_4) do |label, holiday, usage, average_daily_usage|
+        expect(label).to eq(I18nHelper.holiday(:autumn_half_term))
+        expect(holiday).to eq(holiday_usage[holiday_4].previous_holiday)
+        expect(usage).to eq(holiday_usage[holiday_4].previous_holiday_usage)
+        expect(average_daily_usage).to eq(holiday_usage[holiday_4].previous_holiday_usage.kwh / holiday_usage[holiday_4].previous_holiday.days)
+      end
+    end
+  end
+
+  describe '.comparison_row' do
+    context 'when comparison can be made' do
+      it 'yields expected values' do
+        component.comparison_row(holiday_3) do |label, holiday, usage, previous_holiday, previous_holiday_usage|
+          expect(label).to eq('')
+          expect(holiday).to eq(holiday_3)
+          expect(usage).to eq(holiday_usage[holiday_3].usage)
+          expect(previous_holiday_usage).to eq(holiday_usage[holiday_3].previous_holiday_usage)
+          expect(previous_holiday).to eq(holiday_usage[holiday_3].previous_holiday)
+        end
+      end
+    end
+
+    context 'when comparison cant be made as only partial data for holiday' do
+      let(:data_end_date) { final_holiday_end - 2.days }
+
+      it 'returns nil' do
+        expect(component.comparison_row(holiday_4)).to be false
+      end
+    end
+  end
+
+  context 'when rendering' do
```

Comment:
wonder if it would be useful check the value of something like `html.css('tr').map { |row| row.css('th, td').map { |td| td.text.strip }` to check the whole table